### DISCUSSION TITLE
CalDAV Import/Export Synchronization

### DISCUSSION THREAD

**Discussion by ghost** [JordanKnott/taskcafe#80] [Ideas]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...].
Currently use CalDAV to store tasks from other apps (i.e, Tasks.org,etc) however I don't have a front-end to manage tasks on the web-browser via the kanban-like functionality that taskcafe currently offers. Excited to see how the roadmap progresses and wanted to know if CalDAV synchronization and import/export to CALDAV compatible systems is currently on the horizon.

**Describe the solution you'd like**
The ability to import/export and/or sync to CALDAV tasks from NextCloud or other CalDAV-compatible systems manageable through the taskcafe front-end

**Describe alternatives you've considered**
I was thinking about manually exporting tasks from CalDAV and then fitting them to taskcafe task structures but not sure what the final data model for taskcafe would look like in the future
. 
**Additional context**
Using Tasks.org to manage task tracking for now and would like to see if this feature would be something to be integrated into the roadmap moving forward.

**Comment by almereyda** [comment]
This is a very neat idea to leverage the often underused CalDAV tasks and to build a decoupled Kanban frontend for them.

Please note that Nextcloud also has the Cards App, which integrates with the rest of the system. Eventually that could also be a good place to ask for implementing this feature?

**Reply by tedstriker** [reply]
Does nextcloud come with CalDAV support? If so, it would become compatible when the main request is implemented. If not, a topic dedicated to your request would make progress tracking and discussions way easier than mixing it into topic @almereyda

**Reply by almereyda** [reply]
What I was trying to point out is, that Nextcloud allows subscribing to remote calendars like iCal/ICS or CalDAV feeds already, so importing cards with dates via an ICS export of Taskcafé could work out. The other way round appears harder.

CalDAV feeds are usually not shared publicly, which means one needs to authenticate against their endpoints.

While this idea of visualising and handling CalDAV tasks in a Kanban board seems very promising and exciting even, I'm not so sure if this edge case deserves a lot of attention before a Taskcafé 1.0.0 release.

My comment answered to @yairark's idea and did not involve Taskcafé directly.

**Reply by ghost** [reply]
Thanks for the reply @almereyda! Definitely would be a cool feature for future releases!

**Reply by almereyda** [reply]
As a follow up, in the meantime you might wish to have a look into https://github.com/alainm23/planner/issues/555 - if they consider to and succeed in mapping their task sections (which are kanban columns in another view) to some CalDAV metadata, you would have a Kanban interface based on a CalDAV remote.
